store/store.go

```diff
@@ -225,7 +230,7 @@ func (bs *BlockStore) LoadBlockCommit(height int64) *types.Commit {
 // a new block at `height + 1` that includes this commit in its block.LastCommit.
 func (bs *BlockStore) LoadSeenCommit(height int64) *types.Commit {
 	var pbc = new(tmproto.Commit)
-	bz, err := bs.db.Get(calcSeenCommitKey(height))
+	bz, err := bs.db.Get(seenCommitKey(height))
```

Comment:
Investigating into this panic that's started to appear:
```
panic: error reading block seen commit: proto: wrong wireType = 2 for field Height

goroutine 1748 [running]:
github.com/tendermint/tendermint/store.(*BlockStore).LoadSeenCommit(0xc0010e4fe0, 0x1, 0x1908d80)
	/Users/callum/Developer/go/src/github.com/tendermint/tendermint/store/store.go:242 +0x2d0
github.com/tendermint/tendermint/consensus.(*State).LoadCommit(0xc0001c8000, 0x1, 0x0)
	/Users/callum/Developer/go/src/github.com/tendermint/tendermint/consensus/state.go:298 +0xaa
github.com/tendermint/tendermint/consensus.(*Reactor).queryMaj23Routine(0xc00107b780, 0x1c98c60, 0xc00171a000, 0xc0004500d0)
	/Users/callum/Developer/go/src/github.com/tendermint/tendermint/consensus/reactor.go:835 +0x285
created by github.com/tendermint/tendermint/consensus.(*Reactor).AddPeer
	/Users/callum/Developer/go/src/github.com/tendermint/tendermint/consensus/reactor.go:195 +0x151
FAIL	github.com/tendermint/tendermint/consensus	1.657s
```
If anyone has an idea what could have caused it then let me know

Reply:
Sounds like you're loading data into a different Protobuf than what was saved. E.g. commit and SeenCommit are not the same thing (I think).

Reply:
I found out what's happening. The state store and block store were (at least in the tests) using the same db. Since the prefixes overlap, they were overriding each others values

Reply:
lol 
```
blockDB := dbm.NewMemDB()
stateDB := blockDB
```

Reply:
We may want to use unique prefixes across all keys, even though they are saved in the same database. Prevents confusion, and bugs like these.

Reply:
As a side idea, shall I add a little note in the protobuf conversion error saying that this may have been caused by trying to read a different data structure than what was expected or something along the lines of that.

Reply:
I don't think that's essential (to me this is fairly obvious), but feel free to add one if you think it'd be useful.